static/js/components/NewGroupUserForm.jsx

```diff
@@ -1,52 +1,105 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
-import { useDispatch } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
+import TextField from '@material-ui/core/TextField';
+import Autocomplete, { createFilterOptions } from '@material-ui/lab/Autocomplete';
 
-import * as Action from '../ducks/groups';
+import * as groupsActions from '../ducks/groups';
+import * as usersActions from '../ducks/users';
+
+
+const filter = createFilterOptions();
 
 const NewGroupUserForm = ({ group_id }) => {
   const dispatch = useDispatch();
-  const [formState, setState] = useState({
-    newUserEmail: "",
+  const { allUsers } = useSelector((state) => state.users);
+  const [formState, setFormState] = useState({
+    newUserEmail: null,
     admin: false
   });
 
-  const handleChange = (event) => {
-    setState({
-      ...formState,
-      newUserEmail: event.target.value
-    });
-  };
+  useEffect(() => {
+    if (allUsers.length === 0) {
+      dispatch(usersActions.fetchUsers());
+    }
+  }, [dispatch, allUsers]);
 
   const handleSubmit = (event) => {
     event.preventDefault();
-    dispatch(Action.addGroupUser({
+    dispatch(groupsActions.addGroupUser({
       username: formState.newUserEmail,
       admin: formState.admin,
       group_id
     }));
-    setState({
-      newUserEmail: "",
+    setFormState({
+      newUserEmail: null,
       admin: false
     });
   };
 
   const toggleAdmin = (event) => {
-    setState({
+    setFormState({
       ...formState,
       admin: event.target.checked
     });
   };
 
   return (
     <div>
-      <input
-        type="text"
+      <Autocomplete
         id="newUserEmail"
         value={formState.newUserEmail}
-        onChange={handleChange}
+        onChange={(event, newValue) => {
+          if (typeof newValue === 'string') {
+            setFormState({
+              newUserEmail: newValue,
+            });
+          } else if (newValue && newValue.inputValue) {
```

Comment:
what does this else clause handle?

Reply:
As the comment below describes, it handles the case where a new value (not in the options list) is being provided by the user (i.e. they're specifying a _new_ username)